btf/marshal.go

```diff
@@ -200,11 +200,17 @@ func (e *encoder) deflateType(typ Type) (err error) {
 		raw.data, err = e.convertMembers(&raw.btfType, v.Members)
 
 	case *Enum:
-		raw.SetKind(kindEnum)
 		raw.SetSize(v.size())
 		raw.SetVlen(len(v.Values))
 		raw.SetSigned(v.Signed)
-		raw.data, err = e.deflateEnumValues(v.Values)
```

Comment:
Another option would be to return `(any, error)` from deflateEnumValues and handle the enum / 64 details there. This is fine too.